File: base/src/main/java/me/juhezi/module/base/BaseApplication.java
package me.juhezi.module.base;

import android.app.Application;

import com.juhezi.module.router_annotation.annotation.Proxy;

import me.juhezi.module.router_api.Router;

/**
 * Created by Juhezi[juhezix@163.com] on 2017/8/29.
 */
@Proxy
public class BaseApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        Router.inject(this);
    }
}

File: base/src/main/java/me/juhezi/module/base/BaseApplication.kt
package me.juhezi.module.base

import android.app.Application

import me.juhezi.module.base.router.BaseRouterProxy
import me.juhezi.module.router_api.Router

/**
 * Created by Juhezi[juhezix@163.com] on 2017/8/29.
 */
open class BaseApplication : Application() {

    companion object {
        private lateinit var sInstance: BaseApplication
        @JvmStatic
        fun getInstance() = sInstance
    }

    override fun onCreate() {
        super.onCreate()
        sInstance = this
        onInject()
    }

    open fun onInject() {
        Router.inject(BaseRouterProxy())
    }

}
